examples/fabric/byot/trainer.py

```diff
@@ -0,0 +1,409 @@
+import os
+from collections.abc import Mapping
+from functools import partial
+from typing import Any, cast, List, Literal, Optional, Tuple, Union
+
+import torch
+from lightning_utilities.core import is_overridden
+from lightning_utilities.core.apply_func import apply_to_collection
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from lightning.fabric.fabric import (
+    _PLUGIN_INPUT,
+    _PRECISION_INPUT,
+    _unwrap_objects,
+    Accelerator,
+    Fabric,
+    Logger,
+    Strategy,
+)
+from lightning.fabric.strategies.fsdp import FSDPStrategy
+from lightning.fabric.utilities.rank_zero import rank_zero_warn
+from lightning.fabric.utilities.types import LRScheduler, Optimizable
+from lightning.pytorch.core.module import LightningModule
+
+
+class Trainer:
```

Comment:
Can we use a different name to make it clear that this isnt the real Trainer?

Reply:
What do you suggest? `SimpleTrainer`?

Reply:
`FabricTrainer` or `TrainerWithFabric`. Open to suggestions.

We want to avoid Simple for the same reasons Lite was removed